packages/react-dom/src/__tests__/ReactDOMFizzServer-test.js

```diff
@@ -87,6 +88,22 @@ describe('ReactDOMFizzServer', () => {
       hasErrored = true;
       fatalError = error;
     });
+
+    renderOptions = {};
+    if (gate(flags => flags.enableFizzNoScriptExecution)) {
+      window.__init_instruction_observer__ = () => {
+        global.testDocument = document;
+        global.MutationObserver = jsdom.window.MutationObserver;
+        reactInstructionObserver = ReactDOMFizzServer.getReactDOMClientMutationObserver();
```

Comment:
Like I mentioned below, we should simulate what would happen when we inject this as a `<script src="...">` tag. Not getting an already initialized version of this file.

Once you add the new bundle you can read it with `fs.readFileSync(require.resolve(script.src))`.